x-pack/plugins/maps/public/actions/layer_actions.ts

```diff
@@ -193,31 +227,17 @@ export function setSelectedLayer(layerId: string | null) {
   };
 }
 
-export function removeTransientLayer() {
+export function setSelectedLayerToFirstPreviewLayer() {
```

Comment:
This is a tough one to name. As it's currently written I thought we were promoting a selected layer to preview status, which of course isn't the case. Maybe change the name of this to something like: `setFirstPreviewLayerToSelectedLayer`